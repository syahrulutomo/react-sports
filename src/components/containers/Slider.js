import React, { useState } from 'react';
import PropTypes from 'prop-types';
import prevBtn from '../../assets/images/keyboard_arrow_left-black-18dp.svg';
import nextBtn from '../../assets/images/keyboard_arrow_right-black-18dp.svg'


export const Slider = (props) => {
  const { data, width, space, style } = props;

  const [step, setStep] = useState(0);

  const handlePrev = () => {
    if (step > 0) setStep(step - 1);
  }

  const handleNext = () => {
    if (step + 3 < data.length) setStep(step + 1);
  }

  return (
    <section className="app-slider" style={style}>
      {
        step > 0 ?
        (
          <div className="app-slider--prev" onClick={handlePrev}>
            <img src={prevBtn} alt="prev"/>
          </div>
        ) : ''
      }
      <div className="app-slider--content" 
        style={{ 
          transition: 'transform 1s ease', 
          transform: `translateX(${step * - width}px)`, 
          width: `calc((${width}px *  ${data.length}) + ${space}px * ${data.length -1})`,
        }}
        >
        { data }
      </div>
      {
        step + 3 < data.length ?
        (
          <div className="app-slider--next" onClick={handleNext}>
            <img src={nextBtn} alt="next"/>
          </div>
        ) : ''
      }
      
    </section>
  )
};

Slider.propTypes = {
  data: PropTypes.array.isRequired,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  space: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  handleNext: PropTypes.func,
  handlePrev: PropTypes.func
}


