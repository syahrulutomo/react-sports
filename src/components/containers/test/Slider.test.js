import React from 'react';
import {cleanup} from '@testing-library/react';
import { Slider } from '../Slider'
import ShallowRenderer from 'react-test-renderer/shallow';

afterEach(cleanup)

const data = [
  { a: 1}, { b: 1}, { c: 1}, { d: 1},
];

it('renders correctly Slider component', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Slider data={data} />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});