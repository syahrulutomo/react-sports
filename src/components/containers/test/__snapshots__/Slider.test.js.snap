// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Slider component 1`] = `
<section
  className="app-slider"
>
  
  <div
    className="app-slider--content"
    style={
      Object {
        "transform": "translateX(NaNpx)",
        "transition": "transform 1s ease",
        "width": "calc((undefinedpx *  4) + undefinedpx * 3)",
      }
    }
  >
    Object {
      "a": 1,
    }
    Object {
      "b": 1,
    }
    Object {
      "c": 1,
    }
    Object {
      "d": 1,
    }
  </div>
  <div
    className="app-slider--next"
    onClick={[Function]}
  >
    <img
      alt="next"
      src="keyboard_arrow_right-black-18dp.svg"
    />
  </div>
</section>
`;
